src/HomeAssistantManager.cpp

```
@@ -884,6 +885,58 @@ void HomeAssistantManager::handleCommand(const char* topic, const char* payload)
         return;
     }
 
+    // tray_update: HA sends full AMS tray state for dashboard display
+    if (strcmp(command, "tray_update") == 0) {
+        StaticJsonDocument<768> trayDoc;
+        if (deserializeJson(trayDoc, payload)) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)